### ISSUE TITLE
neo4j shouldn't run as root

### ISSUE THREAD

**Issue by xenoterracide** [neo4j/docker-neo4j#50]
if neo4j has any exploits this would allow the container to be exploited, including updating neo4js binary, or host other malign software within the container. This also makes it easier to attempt to break out of the jail which has been accomplished in the past.

**Comment by benbc** [comment]
@xenoterracide You are absolutely correct. We are uncomfortable with running Neo4j as root in the container; however we haven't yet come up with a way of doing this that we're happy with. There is a partial implementation of this in #22. The problem is that Neo4j requires an elevated value of `ulimit nofiles`, but that is hard to achieve in a Docker image when not running as root.

We hear your feedback and we'll raise the priority of this work.

**Comment by dmlond** [comment]
as more people move to CoreOS, Atomic, Kubernetes, Openshift, etc. this is going to become more pressing.

**Comment by nikolastankovic** [comment]
We have currently the same problem. By policy our company won't allow as to have root access so we have trouble to use Neo4j in our OpenShift platform. Any information when this issue will be solved? Or is there an option to use Neo4j in a container without root?

**Comment by hpehl** [comment]
Any updates here? I need to run Neo4j in OpenShift and the current image doesn't allow this. I've looked at the existing PRs and workarounds, but they don't seem to work.

**Comment by siposr** [comment]
https://docs.docker.com/engine/reference/commandline/run/#set-ulimits-in-container-ulimit
Docker run (now?) can give appropriate amounts of `ulimit nofiles` without root. Can you revalidate your points on this?

**Comment by kfreytag** [comment]
@hpehl @xenoterracide @dmlond @nikolastankovic We've released a change here, so within our Docker images Neo4j now runs under the user `neo4j`.

**Comment by hpehl** [comment]
Thanks for the update @kfreytag
